xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-authorization/xwiki-platform-security-authorization-api/src/main/java/org/xwiki/security/authorization/cache/internal/DefaultSecurityCache.java

```diff
@@ -725,19 +732,15 @@ private boolean isAlreadyInserted(String key, SecurityEntry entry, Collection<Gr
      */
     private void addEntry(String key, SecurityCacheEntry entry) throws ConflictingInsertionException
     {
-        try {
-            newEntry = entry;
-            cache.set(key, newEntry);
-            if (entry.disposed) {
-                // XWIKI-13746: The added entry have been disposed while being added, meaning that the eviction
-                // triggered by adding the entry has hit the entry itself, so remove it and fail.
-                cache.remove(key);
-                throw new ConflictingInsertionException(String.format(
-                    "The cache entry [%s] with key [%s] has been disposed by another thread while being added.", entry,
-                    key));
-            }
-        } finally {
-            newEntry = null;
+        cache.set(key, entry);
+        this.internalEntries.put(key, entry);
+        if (entry.disposed) {
+            // XWIKI-13746: The added entry have been disposed while being added, meaning that the eviction
+            // triggered by adding the entry has hit the entry itself, so remove it and fail.
+            // TODO: this should be impossible now.
```

Comment:
Good point, so I think it might make sense to keep an exception just in case, but maybe additionally log an error. The reason why this is impossible is that disposing now only happens with an explicit removal operation and this cannot happen in parallel. The cache also cannot trigger this anymore so there is no unknown cache operation that could trigger this. Maybe we could add some more checks in some places to ensure that disposed entries are really never returned anywhere.

Reply:
I've added a dedicated, unchecked exception for that case now so we noticed clearly if this happens for whatever reasons we didn't imagine (the exception should be catched and logged as an error in [`DefaultAuthorizationManager`](https://github.com/xwiki/xwiki-platform/blob/4a7bf51c92a63aea65cb51078142bba0bf0a9bfe/xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-authorization/xwiki-platform-security-authorization-api/src/main/java/org/xwiki/security/authorization/DefaultAuthorizationManager.java#L121)).